### DISCUSSION TITLE
How to apply ICs on the boundary

### DISCUSSION THREAD

**Discussion by mikejwg** [idaholab/moose#16607] [Q&A Modules: Phase field] [answered]
Dear all, 

We plan to simulate the directional solidification in a 2D region.  We took the snow.i as the ref input code for our study.  We assume that  the solidification occur first near the bottom line.  For this purpose, we write the code as follows,
[ICs]
  [./wIC]   
     type = MultiSmoothCircleIC
     invalue = 1
     outvalue = 0
     radius = 0.05
     radius_variation_type=uniform
     radius_variation=0.01
     variable = w
     numbub = 6
     bubspac = 1.0
     boundary = left  
  [../] 
[]
But I found no solidification and no temperature change during simulations. 
How should we edit this section.  Thanks very much.

**Comment by laagesen** [comment, marked as answer]
The snow.i example assumes uniform undercooling in a liquid composed of a pure material, and does not include a fixed temperature gradient as would normally be required to simulate directional solidification. So just be aware that the physics you are trying to simulate may not be captured by this input file. We have some work to develop an alloy directional solidification example which should be coming this year sometime.

As far as inserting nuclei in your initial conditions block, it looks like you are trying to put solid seeds on a certain boundary by specifying `boundary = left`. That option doesn't do what you want. What I would do instead is use `SpecifiedSmoothCircleIC` and manually specify the (x,y) positions where you want each solid seed to be placed. Then run for 1 time step and make sure the solid seeds are where you want them before running the simulation out to longer times.

**Reply by mikejwg** [reply]
Thank you for your reminder. Because I just started working with moose code, I now try to get familiar with it by modifying some of the blocks and parameters in the phase_field module based on some examples in the codes .
